desktop/ui/src/main/java/org/datacleaner/widgets/result/AnalyzerResultFutureSwingRenderer.java

```diff
@@ -0,0 +1,122 @@
+ * or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU Lesser General Public License
+ * for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this distribution; if not, write to:
+ * Free Software Foundation, Inc.
+ * 51 Franklin Street, Fifth Floor
+ * Boston, MA  02110-1301  USA
+ */
+package org.datacleaner.widgets.result;
+
+import javax.inject.Inject;
+import javax.swing.JComponent;
+
+import org.datacleaner.api.AnalyzerResult;
+import org.datacleaner.api.Renderer;
+import org.datacleaner.api.RendererBean;
+import org.datacleaner.api.RendererPrecedence;
+import org.datacleaner.beans.MockAnalyzer;
+import org.datacleaner.connection.Datastore;
+import org.datacleaner.connection.DatastoreCatalog;
+import org.datacleaner.connection.DatastoreConnection;
+import org.datacleaner.connection.SchemaNavigator;
+import org.datacleaner.guice.DCModuleImpl;
+import org.datacleaner.job.builder.AnalysisJobBuilder;
+import org.datacleaner.job.builder.AnalyzerComponentBuilder;
+import org.datacleaner.panels.DCPanel;
+import org.datacleaner.result.AnalyzerResultFuture;
+import org.datacleaner.result.renderer.RendererFactory;
+import org.datacleaner.result.renderer.SwingRenderingFormat;
+import org.datacleaner.util.LookAndFeelManager;
+import org.datacleaner.windows.ResultWindow;
+import org.jdesktop.swingx.JXBusyLabel;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.google.inject.Guice;
+import com.google.inject.Injector;
+
+@RendererBean(SwingRenderingFormat.class)
+public class AnalyzerResultFutureSwingRenderer implements Renderer<AnalyzerResultFuture<? extends AnalyzerResult>, JComponent> {
+    
+    private static final Logger logger = LoggerFactory.getLogger(AnalyzerResultFutureSwingRenderer.class);
+
+    @Inject
+    RendererFactory _rendererFactory;
+    
+    @Override
+    public RendererPrecedence getPrecedence(AnalyzerResultFuture<? extends AnalyzerResult> renderable) {
+        return RendererPrecedence.LOW;
+    }
+
+    @Override
+    public JComponent render(AnalyzerResultFuture<? extends AnalyzerResult> renderable) {
+        final JXBusyLabel busyLabel = new JXBusyLabel();
+        busyLabel.setBusy(true);
+        
+        final DCPanel resultPanel = new DCPanel();
+        resultPanel.add(busyLabel);
+        resultPanel.updateUI();
+        
+        
+        renderable.addListener(new AnalyzerResultFuture.Listener<AnalyzerResult>() {
+
+            @Override
+            public void onSuccess(AnalyzerResult result) {
+                Renderer<? super AnalyzerResult, ? extends JComponent> renderer = _rendererFactory.getRenderer(result, SwingRenderingFormat.class);
+                if (renderer != null) {
+                    JComponent jComponent = renderer.render(result);
+                    resultPanel.add(jComponent);
+                } else {
+                    logger.error("No renderer found for: " + result);
```

Comment:
I guess the user would not see this ... So what should he see? I guess we should have the same behaviour as when ResultWindow cannot find a renderer.